### DISCUSSION TITLE
Can't find PyAEDT method for changing circuit port excitation properties after HFSS import

### DISCUSSION THREAD

**Discussion by em-jim** [ansys/pyaedt#3189] [Q&A]
Created an IronPython script to start.  Now I want to convert this to CPython with PyAEDT.

The problem is I can;t find the PyAEDT method in the EDB API for this.
i.e.
oDesign.ChangePortProperty

Example IronPython script:
#==============================================
oDesign.ChangePortProperty("Port1", 
    [
        "NAME:Port1",
        "IIPortName:="        , "Port1",
        "SymbolType:="        , 0,
        "DoPostProcess:="    , False
    ], 
    [
        [
            "NAME:Properties",
            [
                "NAME:NewProps",
                [
                    "NAME:term",
                    "PropType:="        , "TextProp",
                    "OverridingDef:="    , True,
                    "Value:="        , "Design3"
                ]
            ],
            [
                "NAME:ChangedProps",
                [
                    "NAME:TerminationData",
                    "Value:="        , "Zo"
                ],
                [
                    "NAME:pnum",
                    "Value:="        , "1"
                ],
                [
                    "NAME:EnableNoise",
                    "Value:="        , False
                ],
                [
                    "NAME:noisetemp",
                    "Value:="        , "16.85cel"
![AEDT_circuit_port_imp_from_HFSS_simulation](https://github.com/ansys/pyaedt/assets/121181102/7f91f7cf-f64a-496c-be77-e4ceb176709c)

                ]
            ],
            [
                "NAME:DeletedProps", 
                "rz", 
                "iz"
            ]
        ]
    ])
#==============================================

Please provide suggestions for using PyaEDT to change the circuit port properties to point the port impedance to the imported HFSS solution port impedance.

**Comment by Samuelopez-ansys** [comment]
Hi @em-jim,

It is not possible right now to set this property. I have tested to do it manually and even the recorded script is not working, please could you test it? If it fails, you will need to submit first a defect to Ansys development firts.

If it works, please share the project and script, and we will try to implement it.
